registry/assignments.go

```diff
@@ -29,53 +36,91 @@ func (c *client) GetAssignmentList(blockNumber, assignedChainID *big.Int, scanne
+			for i := int64(0); i < agentCount.Int64(); i++ {
+				agentCalls = append(
+					agentCalls, dispatchMulti.NewCall(new(agentRefAtOutput), "agentRefAt", scannerID, big.NewInt(i)),
+				)
+			}
 
-	_, err = c.multiCaller.CallChunked(opts, 100, agentCalls...)
+			_, err := c.multiCaller.CallChunked(opts, 100, agentCalls...)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	)
 	if err != nil {
 		return nil, fmt.Errorf("agentRefAt calls failed: %v", err)
 	}
 
 	// 3rd eth_call: for each assigned bot, get the assigned scanner counts
 	var numScannersCalls []*multicall.Call
-	for i := 0; i < int(agentCount.Int64()); i++ {
-		numScannersCall := dispatchMulti.NewCall(
-			new(numOutput), "numScannersFor", agentCalls[i].Outputs.(*agentRefAtOutput).AgentId,
-		)
-		numScannersCalls = append(numScannersCalls, numScannersCall)
-	}
-	_, err = c.multiCaller.Call(opts, numScannersCalls...)
+	err = withBackoff(
+		c.ctx, func(ctx context.Context) error {
+			numScannersCalls = make([]*multicall.Call, 0)
+			for i := 0; i < int(agentCount.Int64()); i++ {
+				numScannersCall := dispatchMulti.NewCall(
+					new(numOutput), "numScannersFor", agentCalls[i].Outputs.(*agentRefAtOutput).AgentId,
+				)
+				numScannersCalls = append(numScannersCalls, numScannersCall)
+			}
+			_, err := c.multiCaller.Call(opts, numScannersCalls...)
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	)
 	if err != nil {
 		return nil, fmt.Errorf("numScannersFor calls failed: %v", err)
 	}
 
 	// 4th eth_call: get all assigned scanners to all assigned bots
 	var scannerCalls []*multicall.Call
-	for agentIndex, numScannersCall := range numScannersCalls {
-		agent := agentCalls[agentIndex].Outputs.(*agentRefAtOutput)
-		scannerCount := numScannersCall.Outputs.(*numOutput).Num.Int64()
-		for i := int64(0); i < scannerCount; i++ {
-			scannerCall := dispatchMulti.NewCall(
-				new(scannerRefAtOutput), "scannerRefAt", agent.AgentId, big.NewInt(i),
-			)
-			scannerCalls = append(scannerCalls, scannerCall)
-		}
-	}
-	// the amount of scanners can scale up unexpectedly sometimes so this
-	// chunking is a protection against that
-	_, err = c.multiCaller.CallChunked(opts, 100, scannerCalls...)
+	err = withBackoff(
+		c.ctx, func(ctx context.Context) error {
+			scannerCalls = make([]*multicall.Call, 0)
+			for agentIndex, numScannersCall := range numScannersCalls {
+				agent := agentCalls[agentIndex].Outputs.(*agentRefAtOutput)
+				scannerCount := numScannersCall.Outputs.(*numOutput).Num.Int64()
+				for i := int64(0); i < scannerCount; i++ {
+					scannerCall := dispatchMulti.NewCall(
+						new(scannerRefAtOutput), "scannerRefAt", agent.AgentId, big.NewInt(i),
+					)
+					scannerCalls = append(scannerCalls, scannerCall)
+				}
+			}
+
+			// the amount of scanners can scale up unexpectedly sometimes so this
+			// chunking is a protection against that
+			_, err := c.multiCaller.CallChunked(opts, 100, scannerCalls...)
```

Comment:
It's sufficient if we only wrap `Call` or `CallChunked` methods. The others are just for constructing the calls.